middle_end/flambda2/to_cmm/to_cmm_env.ml

```diff
@@ -783,6 +794,72 @@ let inline_variable ?consider_inlining_effectful_expressions env res var =
           let env = remove_binding env var in
           will_inline_simple env res binding)))
 
+(* Handling of aliases between variables *)
+
+(* Situation: [alias_of] is a `must_inline_once` and [var] is used exactly once
+
+   In this case, we do not want to split the binding in this case, but instead
+   just transfer the binding to the new variable, so that we can decide whether
+   to split it at its effective use (and not here where we rebind it to a
+   used-only-once variable).
+
+   Since `Must_inline_once` bindings (or rather the bound variable) can be
+   arbitrarily deep in the stage stack, it would be too costly to change the
+   whole stack, so we use an alias map on the side instead. *)
+let make_alias env res var alias_of =
+  let inline_once_aliases =
+    Variable.Map.add var alias_of env.inline_once_aliases
+  in
+  let env = { env with inline_once_aliases } in
+  env, res
+
+(* Situation: [alias_of] is a must_inline (once or duplicate), and [var] is used
+   more than once.
+
+   In this case, we want to force splitting of the original binding, and then
+   bind the new variable with a `must_inline` inline status *)
+let split_binding_and_rebind ~num_occurrences_of_var env res var alias_of
+    binding =
```

Comment:
Some of those parameters could likely use labels.

Reply:
I added labels to `~var` and `~alias_of`